sdk/dotnet/Pulumi.Tests/StackTests.cs

```diff
@@ -100,14 +100,22 @@ private async Task<(T, IDictionary<string, object?>)> Run<T>() where T : Stack,
 
             Deployment.Instance = new DeploymentInstance(mock.Object);
 
-            // Act
-            var stack = new T();
-            stack.RegisterPropertyOutputs();
-
-            // Assert
-            Assert.NotNull(outputs);
-            var values = await outputs!.DataTask;
-            return (stack, values.Value);
+            try
+            {
+                // Act
+                var stack = new T();
+                stack.RegisterPropertyOutputs();
+
+                // Assert
+                Assert.NotNull(outputs);
+                var values = await outputs!.DataTask;
+                return (stack, values.Value);
+            }
+            finally
+            {
+                // Reset the instance as other tests expect it to be initially null.
+                Deployment.Instance = null!;
```

Comment:
Could we instead implement `IDisposable` and do this inside the `Dispose()` method, since every test actually expects it to be null anyway? See https://xunit.net/docs/shared-context#constructor.

Reply:
Good idea. Done.